src/backend/distributed/utils/citus_ruleutils.c

```diff
@@ -972,16 +992,23 @@ generate_qualified_relation_name(Oid relid)
 /*
  * AppendOptionListToString converts the option list to its textual format, and
  * appends this text to the given string buffer.
+ *
+ * prefix is prepended to the option list, examples: OPTIONS or WITH
+ * separator separates lists elements, example: "," (comma)
+ * operator separates lists keys and values, typically either NULL or "="
  */
 static void
-AppendOptionListToString(StringInfo stringBuffer, List *optionList)
+AppendOptionListToString(StringInfo stringBuffer, List *optionList,
+						 const char *prefix,
+						 const char *separator,
+						 const char *operator)
```

Comment:
I'm not a huge fan of adding all these parameters. This function is doing something fairly trivial, but now you need to consult the call sites to know what exactly it does. What if we just have separate functions for OPTIONS and WITH since they clearly follow a different syntax?

Reply:
The alternative is to have two function with almost the same content, but it's only about 30 lines each. Let's try that and see if we like it better (in a patch revision coming soon).